src/txn/txn.c

```diff
@@ -344,12 +354,6 @@ __wt_txn_snapshot_save_and_refresh(WT_SESSION_IMPL *session)
     /* Swap the snapshot pointers. */
     __txn_swap_snapshot(&txn->snapshot_data.snapshot, &txn->backup_snapshot_data->snapshot);
 
-    /*
-     * __txn_get_snapshot_int will return without getting the new snapshot if the transaction
-     * already has a snapshot so clear the flag WT_TXN_HAS_SNAPSHOT.
-     */
-    F_CLR(txn, WT_TXN_HAS_SNAPSHOT);
```

Comment:
Why did you introduce this structural change? I *think* I see how it works, but I don't understand how it's related to the core change.

Reply:
@agorrod [Here](https://github.com/wiredtiger/wiredtiger/pull/10121/files#diff-4dcc4b903043c44beb3be3bebd6d1be54eb0b1f68db50925701ac1869c466cb2R214) we expect all the transactions to have this WT_TXN_HAS_SNAPSHOT flag set if it has entered the commit generation at least once.

Also, the change is mainly for code readability purposes because it does not sound good to clear the flag WT_TXN_HAS_SNAPSHOT for a transaction that already has a snapshot and is trying to acquire a new snapshot.

Reply:
So I can be confident I understand, would changing line 214 to:
``` WT_ASSERT(session, F_ISSET(txn, WT_TXN_HAS_SNAPSHOT) || F_ISSET(txn, WT_TXN_REFRESH_SNAPSHOT));```

Have been equivalent?

Reply:
@agorrod A little correction, the assert should be 
` WT_ASSERT(session, F_ISSET(txn, WT_TXN_HAS_SNAPSHOT) || !F_ISSET(txn, WT_TXN_REFRESH_SNAPSHOT));`

Because the flag WT_TXN_REFRESH_SNAPSHOT is set only if WT_TXN_HAS_SNAPSHOT is set [here](https://github.com/wiredtiger/wiredtiger/blob/d3faff25c4e8b1c7708994aa3dfa1b08e1b7575d/src/reconcile/rec_write.c#L300)